src/main/kotlin/com/github/treepat/target_tree/default_tree/DefaultTargetTreeNode.kt

```diff
@@ -11,20 +11,31 @@ class DefaultTargetTreeNode(
     override var children: List<TargetTreeNode> = mutableListOf()
 ) : TargetTreeNode, Comparable<DefaultTargetTreeNode> {
 
-    var parent: TargetTreeNode? = null
+    var posAsChild: Int = -1
+    var parent: DefaultTargetTreeNode? = null
 
-    override fun moveToRightSibling(): TargetTreeNode? = (parent as? DefaultTargetTreeNode)?.getRightSibling(this)
+    override fun moveToRightSibling(): TargetTreeNode? {
+        if (posAsChild != -1 && posAsChild + 1 < parent?.children!!.size) {
+            return this.parent?.children!![posAsChild + 1]
+        }
+        return null
+    }
 
-    override fun moveToLeftSibling(): TargetTreeNode? = (parent as? DefaultTargetTreeNode)?.getLeftSibling(this)
+    override fun moveToLeftSibling(): TargetTreeNode? {
+        if (posAsChild > 0) {
+            return this.parent?.children!![posAsChild - 1]
+        }
+        return null
+    }
```

Comment:
Some changes were made in `DefaultTargetTreeNode` to fix issue #75 but they are very trivial and shouldn't be our focus.